beacon-chain/state/state-native/getters_validator.go

```diff
@@ -413,3 +414,54 @@ func (b *BeaconState) inactivityScoresVal() []uint64 {
 	copy(res, b.inactivityScores)
 	return res
 }
+
+// ActiveBalanceAtIndex returns the active balance for the given validator.
+//
+// Spec definition:
+//
+//	def get_active_balance(state: BeaconState, validator_index: ValidatorIndex) -> Gwei:
+//	    max_effective_balance = get_validator_max_effective_balance(state.validators[validator_index])
+//	    return min(state.balances[validator_index], max_effective_balance)
+func (b *BeaconState) ActiveBalanceAtIndex(i primitives.ValidatorIndex) (uint64, error) {
+	if b.version < version.Electra {
+		return 0, errNotSupported("ActiveBalanceAtIndex", b.version)
+	}
+
+	b.lock.RLock()
+	defer b.lock.RUnlock()
+
+	if i >= primitives.ValidatorIndex(len(b.validators)) {
+		return 0, errors.Wrapf(consensus_types.ErrOutOfBounds, "validator index %d does not exist", i)
+	}
+	v := b.validators[i]
+
+	return min(b.balances[i], helpers.ValidatorMaxEffectiveBalance(v)), nil
```

Comment:
Should we check `b.balances` length? 
```
if i >= primitives.ValidatorIndex(len(b. balances)) {
		return 0, errors.Wrapf(consensus_types.ErrOutOfBounds, "validator index %d does not exist", i)
	}
```

Reply:
Yes

Reply:
Done in d54497a00544a572d979f0c1123ba858135d5ab6